.github/workflows/link-check.yml

```
@@ -0,0 +1,53 @@
+name: Build and check Links
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build_and_serve:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build Next.js App
+        run: yarn build
+
+      - name: Serve App Locally
```

I think you need to remove this steps and add this to linkchecker job as we need our app to run locally so that linkchecker can do it job.
```suggestion
     Remove this step
```

---

Thanks for your review @aialok 

no we need to first build and deploy it here

The deploy step is missing here as the deployment is not yet clear

@aialok

---

Yes, I too agree with you regarding to need to build and deploy.

---

> Thanks for your review @aialok
> 
> no we need to first build and deploy it here and pass the deployed url in the linkchecker action (i have put localhost:3000 as of now)
> 
> The deploy step is missing here as the deployment is not yet clear
> 
> @aialok

Got it man. But I still have one doubt what do you mean by deploy locally. Deploy locally means running on localhost?

Thank you @utnim2. You have done pretty good job : )